docs/docs/1.0.6/documentation/health.md

```diff
@@ -0,0 +1,160 @@
+lifecycle of the application. 
+
+Kuma supports several aspects of the health checking. There are two policies which allows configuring active and passive:
+[Health Check](../policies/health-check.md) and [Circuit Breaker](../policies/circuit-breaker.md).
+
+Kuma is able to track the status of the Envoy proxy. If grpc stream with Envoy is disconnected then Kuma considers this 
+proxy as offline. 
+
+Also, every `inbound` in the Dataplane model has `health` section:
+
+```yaml
+type: Dataplane
+mesh: default
+name: web-01
+networking:
+  address: 127.0.0.1
+  inbound:
+    - port: 11011
+      servicePort: 11012
+      health:
+        ready: true
+      tags:
+        kuma.io/service: backend
+        kuma.io/protocol: http
+```
+
+This `health.ready` status is intended to show the status of the application itself. It is set differently depending on 
+the environment ([Kubernetes](#kubernetes-probes) or [Universal](#universal-probes)), but it's treated the same way 
+regardless of the environment:
+
+- if proxy status is `Offline`, then Dataplane is `Offline`:
+- if proxy status is `Online`:
+  - if all inbounds are ready then Dataplane is `Online`
+  - if all inbounds are not ready then Dataplane is `Offline`
+  - if at least one of the inbounds is not ready then Dataplane is `Partially degraded` 
+  - if inbound is not ready then it's not included in the load-balancer set which means it doesn't receive the traffic
+  - if all inbounds which implement the same service are ready then service is `Online`
+  - if all inbounds which implement the same service are not ready then service is `Offline`
+  - if at least one of the inbounds which implement the same service is not ready then service is `Partially degraded`
+
+## Kubernetes probes
+
+Kuma natively supports the `httpGet` Kubernetes probes. By default, Kuma overrides the specified probe with a virtual one. For example, if we specify the following probe:
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /metrics
+    port: 3001
+  initialDelaySeconds: 3
+  periodSeconds: 3
+```
+
+Kuma will replace it with:
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /3001/metrics
+    port: 9000
+  initialDelaySeconds: 3
+  periodSeconds: 3
+```
+
+Where `9000` is a default virtual probe port, which can be configured in `kuma-cp.config`:
+
+```yaml
+runtime:
+  kubernetes:
+    injector:
+      virtualProbesPort: 19001
+```
+And can also be overwritten in the Pod's annotations:
+
+```yaml
+annotations:
+  kuma.io/virtual-probes-port: 19001
+```
+
+To disable Kuma's probe virtualziation, we can either set it in Kuma's configuration gile `kuma-cp.config`:
+
+```yaml
+runtime:
+  kubernetes:
+    injector:
+      virtualProbesEnabled: true
+```
+
+or in the Pod's annotations:
+
+```yaml
+annotations:
+  kuma.io/virtual-probes: enabled
+```
+
+:::tip
+Even if virtual probes are disabled Kuma takes `pod.status.containerStatuses.ready` in order to fill `dataplane.inbound.health` section.
```

Comment:
therefore traffic to the pod will be only send when all containers in the pod are ready

Reply:
That should not be a tip, but the first sentence of this section.

Reply:
Fixed, it's not a tip anymore, but the first comment is not entirely true, traffic will be sent based on the health of the particular inbound